Essentials/src/com/earth2me/essentials/commands/Commandsethome.java

```diff
@@ -41,19 +42,41 @@ public void run(final Server server, final User user, final String commandLabel,
                 }
             }
         }
+
         if (checkHomeLimit(user, usersHome, name)) {
             name = "home";
         }
         if ("bed".equals(name) || NumberUtil.isInt(name)) {
             throw new NoSuchFieldException(tl("invalidHomeName"));
         }
+        if (ess.getSettings().isSetSameHomeByConfirm()) {
+            String uuid = user.getBase().getUniqueId().toString();
+            for (String h : usersHome.getHomes()) {
+                if (h.equalsIgnoreCase(name)) {
+                    if (!confirmSetHome.contains(uuid)) {
+                        confirmSetHome.add(uuid);
+
+                        int time = (20 * 10);
+                        ess.getServer().getScheduler().runTaskLater(ess, () -> confirmSetHome.remove(uuid), time);
```

Comment:
This isn’t how you implement cooldowns

Reply:
Why? This is the best way to delay and make less code.

Reply:
alternatively you can use system time for cooldowns

Reply:
I would add a last confirm set home field to the User with the name of the home as well as the time when the initial command was run.

Reply:
Well, the user has already typed the name of the home to be overwritten, so it doesn't need to be done again, just to confirm that the home location is overwritten.

Reply:
Please see below comments. The only requirement with your code is that the user types in _a_ saved home name twice. The code does not lock it to be the same home - these could be different homes.

Reply:
This still needs addressed to be merged.